File: setpuzzle.rb
#!/usr/bin/env ruby
# http://rosettacode.org/wiki/Set_puzzle

def create_deck
	colors = ["red","green","purple"]
	symbols = ["oval","squiggle","diamond"]
	numbers = ["1","2","3"]
	shadings = ["solid","open","striped"]
  deck =[]
  colors.each do |color|
  	symbols.each do |symbol|
  		numbers.each do |number|
  			shadings.each do |shading|
  				deck << [color,symbol,number,shading]
  			end
  		end
  	end
  end
  deck
end

def deal_cards(number)
	create_deck.shuffle[0..number-1]
end	

def show_cards(number)
	p "Dealt #{number} cards"
  deal_cards(number).each { |c| p c[0] + " - " + c[1] + " - " + c[2] + " - " + c[3] }
end

show_cards(9) 	

File: setpuzzle.swift
// http://rosettacode.org/wiki/Set_puzzle
import Darwin

func createDeck() -> [String] {
  let colors = ["red","green","purple"]
  let symbols = ["oval","squiggle","diamond"]
  let numbers = ["1","2","3"]
  let shadings = ["solid","open","striped"]
  var deck: [String] = []
  for color in colors {
    for symbol in symbols {
      for number in numbers {
        for shading in shadings {
          deck.append("\(color) - \(symbol) - \(number) - \(shading)")
        }
      }
    }
  }
  return deck
}

func shuffle(array:[String]) -> [String] {
  let max = array.count
  var mutableArray = array
  for var i = 0; i < max; i++ {
    var rand = Int(arc4random_uniform(UInt32(max)))
    swap(&mutableArray[i],&mutableArray[rand])
  }
  return mutableArray
}

func dealCards(number:Int) -> [String] {
  let cards = Array(shuffle(createDeck())[0...number])
  return cards
}

println(dealCards(4))
